ledger-service/http-json/search-query-language.rst

```diff
@@ -0,0 +1,99 @@
+.. Copyright (c) 2019 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+/contracts/search query language
+================================
+
+The body of ``POST /contracts/search`` looks like so::
+
+  {"%templates": [...template IDs...],
+   ...other query elements...}
+
+The elements of that query are defined here.
+
+Fallback rule
+-------------
+
+Unless otherwise required by one of the other rules below or to follow,
+values are interpreted according to the `LF values' JSON format
+<../lf-value-json/specification.rst>`_, and compared for equality.
+
+Simple equality
+---------------
+
+Match documents having at least all the (potentially nested) keys
+expressed in the query. The result document may contain additional
+properties.
+
+Example: ``{ person: { name: "Bob" }, city: "London" }``
+
+- Match: ``{ person: { name: "Bob", dob: "1956-06-21" }, city: "London",
+  createdAt: "2019-04-30T12:34:12Z" }``
+- No match: ``{ person: { name: "Bob" }, city: "Zurich" }``
+- Typecheck failure: ``{ person: { name: ["Bob", "Sue"] }, city:
+  "London" }``
+
+Example: ``{ favorites: ["vanilla", "chocolate"] }``
+
+- Match: ``{ favorites: ["vanilla", "chocolate"] }``
+- No match: ``{ favorites: ["chocolate", "vanilla"] }``
+- No match: ``{ favorites: ["vanilla", "strawberry"] }``
+- No match: ``{ favorites: ["vanilla", "chocolate", "strawberry"] }``
+
+A JSON object, when considered with a record type, is always interpreted
+as a field equality query. Its type context is thus mutually exclusive
+with `the forthcoming comparison queries
```

Comment:
Can you add an example here?

Reply:
No, because

1. this is a statement that _no example of x exists_, and
2. to exemplify indirectly (e.g. by saying "this is always left, and this is always right", though I am skeptical of the value of such an inclusion) would require the full definition of comparison queries as meaningful context, which does not belong in this document until #2780.